File: pyo3async/tests/simple.rs
//! file: simple.rs
//! author: Jacob Xie
//! date: 2023/09/08 23:04:52 Friday
//! brief:

use pyo3::prelude::*;

// simple test case
#[pyo3_asyncio::tokio::main]
async fn simple() -> PyResult<()> {
    let fut = Python::with_gil(|py| {
        let asyncio = py.import("asyncio")?;

        pyo3_asyncio::tokio::into_future(asyncio.call_method1("sleep", (1.into_py(py),))?)
    })?;

    fut.await?;

    Ok(())
}

///////////////////////////////////////////////////////////////////////////////////////////////////

File: eigen-demo/simple.cc
/**
 * @file:	simple.cc
 * @author:	Jacob Xie
 * @date:	2023/12/22 22:49:20 Friday
 * @brief:
 *
 * compile:
 * g++ -I /opt/eigen-3.4.0 simple.cc -o simple
 **/

#include <Eigen/Dense>
#include <iostream>

using Eigen::MatrixXd;

int main(int argc, char** argv)
{
  MatrixXd m(2, 2);
  m(0, 0) = 3;
  m(1, 0) = 2.5;
  m(0, 1) = -1;
  m(1, 1) = m(1, 0) + m(0, 1);

  std::cout << m << std::endl;

  return 0;
}
